src/ServiceControl.Infrastructure.Metrics/Counter.cs

```diff
@@ -7,14 +7,16 @@
     public class Counter
     {
         readonly string name;
+        readonly Func<bool> enabled;
         readonly int[] eventsPerSecond;
         readonly int[] movingAverage;
         readonly long[] movingAverageEpochs;
         long epoch;
 
-        public Counter(string name)
+        internal Counter(string name, Func<bool> enabled)
```

Comment:
Do we support hot reloading of configuration? If No I'd say it would be better to just make this a bool. Otherwise we will always have a closure allocation due to the indirection

Reply:
👍 I fixed it.